test/parallel_api/ranges/std_ranges_reverse.pass.cpp

```diff
@@ -0,0 +1,34 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    //A checker below modifies a return type; a range based version with policy has another return type.
+    auto reverse_checker = [](auto&&... args) {
+        return std::ranges::reverse(std::forward<decltype(args)>(args)...);
+    };
```

Comment:
I do not quite understand why and how the return type is changed.

`std::ranges::reverse`, when fed with a range, returns `ranges::borrowed_iterator_t<R>`. The same type is returned by the algorithm with the policy. 

Is that checker really needed?

Reply:
I guess, the comment is not relevant.

Reply:
I suppose that the checker can be removed as well.